src/features/workout-builder/ui/workout-stepper.tsx

```diff
@@ -261,6 +263,12 @@ export function WorkoutStepper() {
         onStartWorkout={handleStartWorkout}
         totalSteps={STEPPER_STEPS.length}
       />
+      
+      <AddExerciseModal
+        isOpen={showAddExerciseModal}
+        onClose={() => setShowAddExerciseModal(false)}
```

Comment:
you can leverage `useBoolean` as in `src/features/workout-builder/ui/exercise-list-item.tsx`